### ISSUE TITLE
JavadocParser public class?

### ISSUE THREAD

**Issue by sniederm** [javaparser/javaparser#3211]
Shouldn't JavadocParser be a public class or did I miss something how to use it correctly?

https://github.com/javaparser/javaparser/blob/master/javaparser-core/src/main/java/com/github/javaparser/JavadocParser.java 

BTW: Great work, keep going!

**Comment by matozoid** [comment]
Oh, I remember this one. Since the StaticJavaParser class acts as the main entrypoint for javaparser-core, it also has the entrypoint for parsing Javadoc: `StaticJavaParser.parseJavadoc`. If you reason that way, JavadocParser.parse(String) should *not* be public, and JavadocParser.parse(JavadocComment), hmmm, I thought that could be called from JavadocComment nodes, but it seems to be unused?

Okay, those were the design decisions in 2017. The current team can decide if those are still okay :-)

**Comment by MysterAitch** [comment]
Not one I have an opinion on currently.

I know that there was work on the javadoc parser quite a while back but it mostly passed me by as I don't use it.

**Comment by matozoid** [comment]
Yes, that was good work that needs to be picked up again, but it was far from finished.
